grails-app/migrations/0.9.x/changelog-2023-09-08-1340-remove-missing-price-fk-on-product-packages.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog/1.9"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog/1.9 http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
+
+  <changeSet author="drodzewicz" id="080920231340-0">
+    <preConditions onFail="MARK_RAN">
+      <not>
+        <sqlCheck expectedResult="0">
+          SELECT COUNT(*) FROM product_package ppkg
+          LEFT JOIN product_price pp
+          ON ppkg.product_price_id=pp.id
```

Comment:
These two lines make no sense to me `ON ppkg.product_price_id=pp.id` and `WHERE ppkg.product_price_id IS NOT NULL AND pp.id IS NULL` it looks like they are mutually excluding

Reply:
Yes, I understand your confusion, but this is exactly the bug the lack of deep validation caused.
Let me explain it with an example

When creating a ProductSupplier, even though there was a validation error on ProductPrice, it still created ProductSupplier and ProductPackage. It also created a foreign key on ProductPackage pointing to ProductPrice, but did not persist that ProductPrice to the database.
So the FK was pointing to nothing, to a non existing record.

so executing this SQL statement
```SQL
SELECT ppkg.id, ppkg.product_price_id, pp.id FROM product_package ppkg
LEFT JOIN product_price pp
ON ppkg.product_price_id=pp.id
WHERE ppkg.product_price_id IS NOT NULL AND pp.id IS NULL;
```

will return such a result
![image](https://github.com/openboxes/openboxes/assets/35233347/9c105ab9-6374-4841-8e74-e2195fa27ef4)